### PR TITLE
issue-2055 do not try to reindex deleted resources

### PR DESCRIPTION
**Pull request by punktilious**
Signed-off-by: Robin Arnold <robin.arnold23@ibm.com>

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java [hunk 1]
```diff
@@ -1926,33 +1926,48 @@ public int reindex(FHIRPersistenceContext context, OperationOutcome.Builder oper
                     logicalId = parts[1];
                 }
 
-                // Look up the resourceTypeId for the given resourceType
+                // Look up the optional resourceTypeId for the given resourceType parameter
```
Comment (lmsurpre) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java:1929]: what does it mean for the resourceTypeId to be optional?

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgres/PostgresReindexResourceDAO.java [hunk 2]
```diff
@@ -28,11 +28,14 @@
 import com.ibm.fhir.persistence.jdbc.impl.ParameterTransactionDataImpl;
 
 /**
- * Derby specialization of the DAO used to assist the reindex custom operation
+ * PostgreSQL specialization of the DAO used to assist the reindex custom operation
```
Comment (lmsurpre) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgres/PostgresReindexResourceDAO.java:31]: good catch :-)

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java (hunk 1 above)
Comment (punktilious) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java:1929]: You can reindex:
1 all resource types
2 "Observation" - all Observation resources
3 "Observation/abc123" - a specific Observation resource

For 1, the resourceTypeId will be null.

**Pull request merged**